### PR TITLE
Multidimentional items

### PR DESCRIPTION
**Pull request by ShishkinDmitriy**
Initial draft for multidimentional items.
The idea is that item can have 0..N controls together.
Maybe item level should keep simple and move specific to controls.
It allows to combine `int+float`, `int+int+int` or `float+string` control in single item.
Previous items implementations become to controls, for exaple `ItemList` -> `MenuItem + ControlList` and so on.

https://github.com/user-attachments/assets/02e7f0d0-2918-4d1a-b028-f8c428bd24b9

```c++
MENU_SCREEN(mainScreen, mainItems,
    new ItemControl<float>("Temp", 
        new NumberControl<float>({.min=-30.0, .max=40.0, .initial=26.5, .step=0.5, .format="%+2.1f\002C", .cycle=false, .blinkerOffset=2}),
        cb1),
    new ItemControl<float>("Price", 
        new NumberControl<float>({.min=0.0, .max=10.0, .initial=1.0, .step=0.1, .format="$%2.2f", .cycle=true}),
        cb1),
    new ItemControl<float, int>("Dist", 
        new NumberControl<float>({.min=0.0, .max=10.0, .initial=1.0, .step=0.1, .format="%2.1fm", .cycle=false, .blinkerOffset=1}),
        new NumberControl<int>({.min=0, .max=100, .initial=10, .step=1, .format="\003%d%%", .cycle=false, .blinkerOffset=1}),
        cb2),
    new ItemControl<int, int, int>("Time", 
        new NumberControl<int>({.min=0, .max=23, .initial=0, .step=1, .format="%02d", .cycle=true}),
        new NumberControl<int>({.min=0, .max=59, .initial=0, .step=1, .format=":%02d", .cycle=true}),
        new NumberControl<int>({.min=0, .max=59, .initial=0, .step=1, .format=":%02d", .cycle=true}),
        cb3));
```

For now it's not stable, so it's draft, need to check overflow of line length.

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -19,4 +19,5 @@ lib_deps =
 	feilipu/FreeRTOS@^11.0.1-5
 	mprograms/SimpleRotary@^1.1.3
 	madleech/Button@^1.0.0
-	Wire
\ No newline at end of file
+	Wire
+	https://github.com/mpaland/printf.git#v4.0.0
```
Comment (forntoh) [inline platformio.ini:23]: Examples won't compile unless you add this dependency in the `compile-arduino.yml` workflow

File: src/display/LiquidCrystalAdapter.h [hunk 2]
```diff
@@ -26,6 +26,10 @@ class LiquidCrystalAdapter : public DisplayInterface {
         lcd->clear();
         lcd->createChar(0, upArrow);
         lcd->createChar(1, downArrow);
+        byte celsius[8] = {B11100, B10100, B11100, B0000, B00000, B00000, B00000, B00000};
+        byte plusMinus[] = {B00000, B00100, B01110, B00100, B00000, B01110, B00000, B00000};
+        lcd->createChar(2, celsius);
+        lcd->createChar(3, plusMinus);
```
Comment (forntoh) [inline src/display/LiquidCrystalAdapter.h:32]: These can be created outside the display adapter (in `setup()`) as you have the lcd instance there, doesn't really need to be here.

Another option to make it maybe easy for use, is to extract them into another utility class eg `DisplayConfig`, in the `DisplayConfig` for LCD screens for example, one can create the custom chars outside and pass them into the display interface through the `DisplayConfig`. In the example sketch, you would pass `\002` and `\003` and they'll be created. We can also use it for customising the indicator arrows and the other custom chars that exist in the LCD adapters. The option of using this class would be maybe better as the dev wouldn't know which bytes have already been taken unless the dive deep into the code.

**Comment by ShishkinDmitriy** [comment]
MenuItem's hierarchy:

MenuItems

So for different items we can use composition of Controls/Widgets instead of inheritance, should be mo flexible.

File: .github/workflows/compile-arduino.yml [hunk 3]
```diff
@@ -33,6 +33,7 @@ jobs:
             - name: FreeRTOS
             - name: SimpleRotary
             - name: LiquidCrystal I2C
+            - name: printf
```
Comment (forntoh) [inline .github/workflows/compile-arduino.yml:36]:
```suggestion
            - source-url: https://github.com/mpaland/printf.git
              version: latest
```

Comment (ShishkinDmitriy) [reply .github/workflows/compile-arduino.yml:36]: This library is needed only for `float` ssupport, suppose we can implement flat specialization of control manually, get existing implementation.

**Comment by forntoh** [comment]
You can change your target branch to `master` and re-open the PR

**Comment by ShishkinDmitriy** [comment]
@forntoh Can you kindly create new target branch for me?)

**Comment by forntoh** [comment]
You can target `master`, the only reason why we split the other one was because it had so many changes (especially breaking ones), this change is just to include a new item type, right!

**Comment by forntoh** [comment]
FYI @ShishkinDmitriy 

> For now it's not stable, so it's draft, need to check overflow of line length.

Fixed in #233, you can use it as your new target.

**Pull request closed without merging**